### PR TITLE
GnuCOBOL, new recipe

### PR DESCRIPTION
**Pull request by Knezzen**

### CODE REVIEW

File: dev-lang/cobol/gnucobol-3.0~rc1.recipe [hunk 1]
```diff
@@ -0,0 +1,70 @@
+SUMMARY="A free COBOL compiler"
+DESCRIPTION="GnuCOBOL (formerly OpenCOBOL) is a free COBOL compiler."
+HOMEPAGE="https://sourceforge.net/projects/open-cobol/"
```
Comment (extrowerk) [inline dev-lang/cobol/gnucobol-3.0~rc1.recipe]: Ordering: https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#ordering

File: dev-lang/cobol/gnucobol-3.0~rc1.recipe [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	cmd:m4
+	cmd:perl
+	lib:libdb
+	lib:libgmp
+	lib:libidn2
+	lib:libintl
+	lib:libltdl
+	lib:libncurses
+	"
+BUILD_REQUIRES="
```
Comment (extrowerk) [inline dev-lang/cobol/gnucobol-3.0~rc1.recipe]: Please add a newline before this

File: dev-lang/cobol/gnucobol-3.0~rc1.recipe [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	cmd:autoconf
+	cmd:bison
+	cmd:flex
+	cmd:gcc
+	cmd:make
+	cmd:m4
+	cmd:perl
+	cmd:pkg_config
+	"
+BUILD()
```
Comment (extrowerk) [inline dev-lang/cobol/gnucobol-3.0~rc1.recipe]: Please add a newline before this

File: dev-lang/cobol/gnucobol-3.0~rc1.recipe [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+{
+	autoreconf -vfi
+	runConfigure ./configure
+	make $jobArgs
+}
+INSTALL()
```
Comment (extrowerk) [inline dev-lang/cobol/gnucobol-3.0~rc1.recipe]: Please add a newline before this

File: dev-lang/cobol/gnucobol-3.0~rc1.recipe [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+{
+	make install
+	prepareInstalledDevelLib libcob
+	rm $developLibDir/*.la
+	fixPkgconfig
+
+	# devel package
+	packageEntries devel \
+		$developDir
+}
```
Comment (extrowerk) [inline dev-lang/cobol/gnucobol-3.0~rc1.recipe]: Could you please check if it comes with some tests? If so, please add a TEST() section too. Do not forget about the newline before the TEST()

File: dev-lang/cobol/gnucobol-3.0~rc1.recipe [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+DESCRIPTION="GnuCOBOL (formerly OpenCOBOL) is a free COBOL compiler."
+HOMEPAGE="https://sourceforge.net/projects/open-cobol/"
+SOURCE_URI="https://downloads.sf.net/open-cobol/gnu-cobol/gnucobol-3.0-rc1.tar.gz"
+REVISION="1"
+LICENSE="GNU GPL v3"
+COPYRIGHT="2001-2019 Free Software Foundation, Inc"
+CHECKSUM_SHA256="e55aeea6b1f77e763b4cd4b520c78eb06da7671b4dcc76463fd0237dbf2e4816"
+SOURCE_DIR="gnucobol-3.0-rc1"
+
+ARCHITECTURES="x86_64"
```
Comment (extrowerk) [inline dev-lang/cobol/gnucobol-3.0~rc1.recipe]: Could you please test it on x86? You will have to add some $secondaryArchSuffix-es here and there.

File: dev-lang/cobol/gnucobol-3.0~rc1.recipe (hunk 5 above)
Comment (Knezzen) [reply dev-lang/cobol/gnucobol-3.0~rc1.recipe]: No tests that I know of. Updated the rest as you commented.

File: dev-lang/cobol/gnucobol-3.0~rc1.recipe (hunk 6 above)
Comment (Knezzen) [reply dev-lang/cobol/gnucobol-3.0~rc1.recipe]: I don't have a x86 Haiku machine at the moment, went x86_64 on both my Haiku machines.
I could test it in a VM though.

File: dev-lang/cobol/gnucobol-3.0~rc1.recipe [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+	cmd:cobcrun
+	lib:libcob
+	"
+REQUIRES="
+	haiku
+	cmd:bison
+	cmd:flex
+	cmd:gcc
+	cmd:m4
+	cmd:perl
```
Comment (Begasus) [inline dev-lang/cobol/gnucobol-3.0~rc1.recipe]: I don't these should go in here?
```diff
	cmd:bison
	cmd:flex
	cmd:gcc
	cmd:m4
	cmd:perl
-	cmd:bison
-	cmd:flex
-	cmd:gcc
-	cmd:m4
-	cmd:perl
```

File: dev-lang/cobol/gnucobol-3.0~rc1.recipe [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+SOURCE_DIR="gnucobol-3.0-rc1"
+
+ARCHITECTURES="x86_64"
+
+PROVIDES="
+	gnucobol = $portVersion
+	cmd:cob_config
+	cmd:cobc
+	cmd:cobcrun
+	lib:libcob
```
Comment (Begasus) [inline dev-lang/cobol/gnucobol-3.0~rc1.recipe]: soname is 4.0.0 so it has to be added here (or use libVersionCompat)
It also need a devel:libcob to install the static lib (strange that haikuporter -S doesn't complain about it)

File: dev-lang/cobol/gnucobol-3.0~rc1.recipe [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+REVISION="1"
+SOURCE_URI="https://downloads.sf.net/open-cobol/gnu-cobol/gnucobol-3.0-rc1.tar.gz"
+CHECKSUM_SHA256="e55aeea6b1f77e763b4cd4b520c78eb06da7671b4dcc76463fd0237dbf2e4816"
+SOURCE_DIR="gnucobol-3.0-rc1"
+
+ARCHITECTURES="x86_64"
+
+PROVIDES="
+	gnucobol = $portVersion
+	cmd:cob_config
```
Comment (Begasus) [inline dev-lang/cobol/gnucobol-3.0~rc1.recipe]: This should go into the devel package I think?

File: dev-lang/cobol/gnucobol-3.0~rc1.recipe [hunk 10]
```diff
@@ -0,0 +1,73 @@
+SUMMARY="A free COBOL compiler"
+DESCRIPTION="GnuCOBOL (formerly OpenCOBOL) is a free COBOL compiler."
+HOMEPAGE="https://sourceforge.net/projects/open-cobol/"
+COPYRIGHT="2001-2019 Free Software Foundation, Inc"
```
Comment (Begasus) [inline dev-lang/cobol/gnucobol-3.0~rc1.recipe]: Missing authors (you can see them if you run cobc --version in Terminal)

File: dev-lang/cobol/gnucobol-3.0~rc1.recipe [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+LICENSE="GNU GPL v3"
```
Comment (Begasus) [inline dev-lang/cobol/gnucobol-3.0~rc1.recipe]: Missing 2 licences (check the source folder for copyright files)

File: dev-lang/cobol/gnucobol-3.0~rc1.recipe [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+	devel:libgmp
+	devel:libidn2
+	devel:libintl
+	devel:libltdl
+	devel:libncurses
+	"
+BUILD_PREREQUIRES="
+	cmd:aclocal
+	cmd:autoconf
+	cmd:bison
```
Comment (Begasus) [inline dev-lang/cobol/gnucobol-3.0~rc1.recipe]:
```diff
+	cmd:awk
```

File: dev-lang/cobol/gnucobol-3.0~rc1.recipe [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+	cmd:flex
+	cmd:gcc
+	cmd:make
+	cmd:m4
+	cmd:perl
```
Comment (Begasus) [inline dev-lang/cobol/gnucobol-3.0~rc1.recipe]: Not needed here:
```diff
-	cmd:m4
-	cmd:perl
```

**Comment by Begasus** [comment]
Thanks for the contribution so far, recipe is probably working fine on x86_64 (some changes requested)
It however also builds for _x86 on gcc2h (secondary architecture), so enabling it would be nice, some parts should be added also:
libVersion (check other recipes how they do it)
PROVIDES_devel (for cmd:cob_config and devel:libcob)
Maybe also a defineDebugInfoPackage?

File: dev-lang/cobol/gnucobol-3.0~rc1.recipe [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+CHECKSUM_SHA256="e55aeea6b1f77e763b4cd4b520c78eb06da7671b4dcc76463fd0237dbf2e4816"
+SOURCE_DIR="gnucobol-3.0-rc1"
+
+ARCHITECTURES="x86_64"
+
+PROVIDES="
+	gnucobol = $portVersion
+	cmd:cob_config
+	cmd:cobc
+	cmd:cobcrun
```
Comment (Begasus) [inline dev-lang/cobol/gnucobol-3.0~rc1.recipe]: Add " = $portVersion" to both

File: dev-lang/cobol/gnucobol-3.0~rc1.recipe [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+REQUIRES="
+	haiku
+	cmd:bison
+	cmd:flex
+	cmd:gcc
+	cmd:m4
+	cmd:perl
+	lib:libdb
+	lib:libgmp
+	lib:libidn2
```
Comment (Begasus) [inline dev-lang/cobol/gnucobol-3.0~rc1.recipe]: Not needed:
```diff
-	lib:libidn2
```

File: dev-lang/cobol/gnucobol-3.0~rc1.recipe [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+	lib:libintl
+	lib:libltdl
+	lib:libncurses
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	devel:libdb
+	devel:libgmp
+	devel:libidn2
```
Comment (Begasus) [inline dev-lang/cobol/gnucobol-3.0~rc1.recipe]: Not needed:
```diff
-	devel:libidn2
```

**Comment by Begasus** [comment]
There is a test suite (mind you 901 tests are performed, 2 skipped and 557 failed here)
Regular test can be run with "make check", there is a test for cobol85 to test with "make test" (needs wget, but failed to find libcob.so.4 here), both tests can be run with "make checkall"

**Comment by Begasus** [comment]
@Knezzen As there hasn't been any activity I pushed my changes based on your work to my github, maybe you can fix things here based on that one :)
https://github.com/Begasus/haikuports/commit/1312454b91d1c72c5734499796f6ff6f73572e97

File: dev-lang/cobol/gnucobol-3.0~rc1.recipe [hunk 17]
```diff
@@ -0,0 +1,68 @@
+SUMMARY="A free COBOL compiler"
+DESCRIPTION="GnuCOBOL (formerly OpenCOBOL) is a free COBOL compiler."
+HOMEPAGE="https://sourceforge.net/projects/open-cobol/"
+COPYRIGHT="2001-2019 Free Software Foundation, Inc"
+LICENSE="GNU GPL v3, GNU LGPL v3, "
```
Comment (Begasus) [inline dev-lang/cobol/gnucobol-3.0~rc1.recipe]: Suggesting (travis check fail here):
```diff
-LICENSE="GNU GPL v3, GNU LGPL v3, "
+LICENSE="GNU GPL v3
+	GNU LGPL v3
+	GNU FDL v1.3" (this needs to be copied from the source to directory "licences"
```

**Comment by Begasus** [comment]
@Knezzen any progress on this? Still changes requested if only for x86_64 before we can merge or close this one.

**Comment by Begasus** [comment]
Closing this one as it conflicts with https://github.com/haikuports/haikuports/pull/4386, this version can be build with the recipe from v2.2 also.

**Comment by Begasus** [comment]
PS. thanks for all the work @Knezzen !

**Pull request closed without merging**